files/build_templates/sonic_debian_extension.j2

```diff
@@ -625,6 +620,11 @@ sudo dpkg --root=$FILESYSTEM_ROOT -P {{ debname }}
 {% endfor %}
 {% endif %}
 
+# Remove sshd host keys, and will regenerate on first sshd start. This needs to be
+# done again here because our custom version of sshd is being installed, which
+# will regenerate the sshd host keys.
+sudo rm -f $FILESYSTEM_ROOT/etc/ssh/ssh_host_*_key*
```

Comment:
Why move this statement?

Reply:
When installing openssl/openssh for fips, the ssh key will be generated again, so moving the removal step after all installing steps are complete.

Reply:
As my understanding, in original code, removal step is after all  openssl/openssh installing steps.

If you have a good reason, is it a general bug fix?

Reply:
It is not a bug, there is an additional installation step to install debian packages, the FIPS packages are installed in the step, see https://github.com/Azure/sonic-buildimage/blob/53e5fe6a93b051c8319d51e109d317beb5166e77/files/build_templates/sonic_debian_extension.j2#L598

We move the removal step, because the FIPS packages will generate the ssh keys again.